### PR TITLE
Taxonomy updates

### PR DESCRIPTION
**Pull request by vincenthoogsteder**
Implemented all changes for the 3 latest specs

This branch also contains the updated Website and Docs instrumented with the new Tracker v.0.0.2 (see https://github.com/objectiv/objectiv-analytics/pull/330)

Progress overview:
https://docs.google.com/document/d/1nm5IFQlIx3GSw0XlZMlxIsFSSiUUfyYmkSzQjkN-UYQ/edit

### CODE REVIEW

**Review by sdirosa — COMMENTED** [review]
One suggestion and one idea for easier maintenance of these properties tables. Overall this looks pretty good to me. 

I guess, next, we have to update the Tracking Docs as well, but I can do that while restructuring them for React.

File: docs/docs/taxonomy/reference/abstract-contexts/AbstractLocationContext.md [hunk 1]
```diff
@@ -7,27 +7,31 @@ title: AbstractLocationContext
 
 import Mermaid from '@theme/Mermaid';
 
-This is the abstract parent of all [Location Contexts](/taxonomy/reference/location-contexts/overview.md). Location Contexts are used to populate a [Tracker's](/tracking/core-concepts/trackers) or 
-[Event's](/tracking/core-concepts/events.md) `location_stack` properties. A Location Stack is meant to describe accurately where an [Event](/tracking/core-concepts/events.md) 
-originated in the UI, e.g. Sections, Menus, etc.
+This is the abstract parent of all [Location Contexts](/taxonomy/reference/location-contexts/overview.md). Location Contexts are used to populate the `location_stack`  properties of an event. A Location Stack is meant to describe accurately where an event originated in the UI.
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/abstract-contexts/AbstractLocationContext.md:10]: Good catch, will update everywhere

File: docs/docs/taxonomy/reference/abstract-contexts/AbstractGlobalContext.md [hunk 2]
```diff
@@ -7,34 +7,38 @@ title: AbstractGlobalContext
...
+    { name: 'SessionContext', to: '/taxonomy/reference/global-contexts/SessionContext' }
 ]}
 />
 
 ### Properties
-All inherited from [AbstractContext](/taxonomy/reference/abstract-contexts/overview.md#abstractcontext).
+|           | type        | description
+| :--       | :--         | :--           
+| **id**    | string      | Unique string to be combined with the Context Type (`_type`) for Context instance uniqueness.
+| **_type** | string      | String literal used during serialization. Should always match the Context interface name.
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/abstract-contexts/AbstractGlobalContext.md:43]: as discussed, we'll leave the inheritance view in the mermaid above. Will keep all inherited property descriptions identical and add notifications for once we have something extra to say on, like automatically tracking PressableContext ID

File: docs/docs/taxonomy/reference/events/FailureEvent.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+| **location_stack**    | array      | The location stack is an ordered list (stack), that contains a hierarchy of location contexts that deterministically describes where an event took place from global to specific. The whole stack (list) is needed to exactly pinpoint where in the UI the event originated.   | [AbstractLocationContext]
+| **global_contexts**    | array      | Global contexts add global / general information about the event. They carry information that is not related to where the Event originated (location), such as device, platform or business data.   | [AbstractGlobalContext]
+| **_type**      | string      | String containing the name of the event type. (eg. ClickEvent).    |   
+| **id**      | string      | Unique identifier for a specific instance of an event. Typically UUID's are a good way of implementing this. On the collector side, events should be unique, this means duplicate id's result in `not ok` events.    | pattern: `^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[a-f0-9]{4}-[a-f0-9]{12}$`    | 
+| **time**      | integer      | Timestamp indicating when the event was generated.    |  
+| **message**      | string      | Failure message    |
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/events/FailureEvent.md:34]: cool, as discussed let's keep that on the list for later

File: docs/docs/taxonomy/reference/overview.md [hunk 4]
```diff
@@ -18,59 +18,44 @@ uniqueness.
...
 * [InteractiveEvent](./events/InteractiveEvent.md)
 * [NonInteractiveEvent](./events/NonInteractiveEvent.md)
+* [PressEvent](./events/PressEvent.md)
 * [SectionHiddenEvent](./events/SectionHiddenEvent.md)
 * [SectionVisibleEvent](./events/SectionVisibleEvent.md)
-* [URLChangeEvent](./events/URLChangeEvent.md)
+* [SuccessEvent](./events/SuccessEvent.md)
 * [VideoEvent](./events/VideoEvent.md)
 * [VideoLoadEvent](./events/VideoLoadEvent.md)
 * [VideoPauseEvent](./events/VideoPauseEvent.md)
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/overview.md:61]: me too!

File: docs/docs/taxonomy/reference/events/NonInteractiveEvent.md [hunk 5]
```diff
@@ -1,40 +1,31 @@
 # NonInteractiveEvent
 
-The parent of [Events](/taxonomy/events) that are not (directly) triggered by an interaction. 
-
-For example, consider the following flow of events:
-
-  1. Press 'play' button in a video player -> ButtonEvent -> InteractiveEvent
-  2. Videoplayer starting playback -> MediaStartEvent -> NonInteractiveEvent
+The parent of [Events](/taxonomy/events) that are not directly triggered by a user action.
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/events/NonInteractiveEvent.md:3]: cool and agreed

File: docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md [hunk 6]
```diff
@@ -21,4 +21,13 @@ import Mermaid from '@theme/Mermaid';
...
+| :--            | :--         | :--           | :--           
+| **location_stack**    | array      | The location stack is an ordered list (stack), that contains a hierarchy of location contexts that deterministically describes where an event took place from global to specific. The whole stack (list) is needed to exactly pinpoint where in the UI the event originated.   | [AbstractLocationContext]
+| **global_contexts**    | array      | Global contexts add global / general information about the event. They carry information that is not related to where the Event originated (location), such as device, platform or business data.   | [AbstractGlobalContext]
+| **_type**      | string      | String containing the name of the event type. (eg. ClickEvent).    |   
+| **id**      | string      | Unique identifier for a specific instance of an event. Typically UUID's are a good way of implementing this. On the collector side, events should be unique, this means duplicate id's result in `not ok` events.    | pattern: `^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[a-f0-9]{4}-[a-f0-9]{12}$`    | 
+| **time**      | integer      | Timestamp indicating when the event was generated.    |
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md:33]: totally agree, going to add notifications about this throughout the taxonomy docs

**Comment by vincenthoogsteder** [comment]
@sdirosa did a round to process your comments and what we discussed, great if you can do a final pass on monday!

**Comment by sdirosa** [comment]
> @sdirosa did a round to process your comments and what we discussed, great if you can do a final pass on monday!

Will do 👌🏼

File: docs/docs/taxonomy/reference/abstract-contexts/AbstractGlobalContext.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdirosa) [reply docs/docs/taxonomy/reference/abstract-contexts/AbstractGlobalContext.md:43]: As long as tables are right after the diagrams I think this works really well

File: docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sdirosa) [reply docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md]: Nice 👌🏼

**Review by sdirosa — APPROVED** [review]
No more comments, let's go!

**Review by sdirosa — APPROVED** [review]
Nice find the DeviceContext docs, totally forgot to check the Docs for that deprecated context

**Comment by vincenthoogsteder** [comment]
@sdirosa did one more sync-up based on schema changes we just did based on your review

**Comment by vincenthoogsteder** [comment]
@sdirosa @jansentom processed the changes we just discussed:
- Abstracts gone from menu and minimal attention
- LocationContext instead of AbstractLocationContext
- SectionContext gone > this will challenge us to see if the LocationContexts are complete. For example we might need to add Header, which we would normally hack into a Section, plus we have no place now to put Main (which we would also name a Section. I actually like that we are forced to make separate LocationContexts out of those, less subjectivity, but let's disucss.
- Simplified menu, overview pages and core concept pages. They should not contain examples and more words than needed.

**Comment by sdirosa** [comment]
> @sdirosa @jansentom processed the changes we just discussed:
> * SectionContext gone > this will challenge us to see if the LocationContexts are complete. For example we might need to add Header, which we would normally hack into a Section, plus we have no place now to put Main (which we would also name a Section. I actually like that we are forced to make separate LocationContexts out of those, less subjectivity, but let's disucss.

I think, for now, we can simply consider those elements as `LocationContext`. It's not an Abstract and ca be used to do so.

Totally agree we should come up with specific Contexts for common elements (section, main, header, footer, etc) but I'd say: only if we have some model to cover them, right?

**Comment by vincenthoogsteder** [comment]
Totally agreed. Feel if we add Main, Header, Footer right now, we are
pretty solid from both UI and Modeling perspective. If we agree will add
those and then run it by Hendrik.

**Comment by vincenthoogsteder** [comment]
@sdirosa fyi did a sweep for simplification and renaming of section-related events

**Comment by vincenthoogsteder** [comment]
@sdirosa @hendrik-obj pls review the changes as discussed today

**Review by borft — APPROVED** [review]
lgtm

File: docs/docs/taxonomy/reference/events/VisibleEvent.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# VisibleEvent
+
+A [NonInteractiveEvent](/taxonomy/reference/events/NonInteractiveEvent.md) that's emitted after a section [LocationContext](/taxonomy/location-contexts/) has become visible.
```
Comment (sdirosa) [inline docs/docs/taxonomy/reference/events/VisibleEvent.md:3]:
```suggestion
A [NonInteractiveEvent](/taxonomy/reference/events/NonInteractiveEvent.md) that's emitted after a [LocationContext](/taxonomy/location-contexts/) has become visible.
```

File: docs/docs/tracking/api-reference/locationTaggers/tagLocation.md [hunk 8]
```diff
@@ -29,7 +29,6 @@ Unless customized via the `options` parameter, the given `instance` determines w
 
 | LocationContext | Automatically triggers
 | :--                                                                                 | :--           
-| [ButtonContext](/taxonomy/reference/location-contexts/ButtonContext.md)                       | [trackClick](/tracking/api-reference/eventTrackers/trackClick.md)
```
Comment (sdirosa) [inline docs/docs/tracking/api-reference/locationTaggers/tagLocation.md:32]: [TODO: surai] Missing PressableContext here

File: docs/docs/tracking/api-reference/locationTaggers/tagLocation.md [hunk 9]
```diff
@@ -29,7 +29,6 @@ Unless customized via the `options` parameter, the given `instance` determines w
...
 | [SectionContext](/taxonomy/reference/location-contexts/SectionContext.md)                     | [trackSectionVisible](/tracking/api-reference/eventTrackers/trackSectionVisible.md), [trackSectionHidden](/tracking/api-reference/eventTrackers/trackSectionHidden.md)
```
Comment (sdirosa) [inline docs/docs/tracking/api-reference/locationTaggers/tagLocation.md:32]: [TODO: surai] Rename to ContentContext

File: docs/docusaurus.config.js [hunk 10]
```diff
@@ -34,7 +34,7 @@ const config = {
   projectName: 'objectiv.io', // Usually your repo name.
 
   onBrokenLinks: 'log',
-  onBrokenMarkdownLinks: 'throw',
+  onBrokenMarkdownLinks: 'log',
```
Comment (sdirosa) [inline docs/docusaurus.config.js:37]: [TODO: surai] Revert this after all links have been fixed (Tracker API Reference links to the Taxonomy)

File: docs/docs/tracking/api-reference/locationTaggers/tagLocation.md (hunk 9 above)
Comment (sdirosa) [reply docs/docs/tracking/api-reference/locationTaggers/tagLocation.md:32]: Also the table is wrong, wrong events

Comment (sdirosa) [reply docs/docs/tracking/api-reference/locationTaggers/tagLocation.md]: Fixed

File: docs/docs/tracking/api-reference/locationTaggers/tagLocation.md (hunk 8 above)
Comment (sdirosa) [reply docs/docs/tracking/api-reference/locationTaggers/tagLocation.md:32]: Added

File: docs/docusaurus.config.js (hunk 10 above)
Comment (sdirosa) [reply docs/docusaurus.config.js]: All links have been fixed, and this has been reverted

File: docs/docs/taxonomy/introduction.md [hunk 11]
```diff
@@ -33,5 +33,5 @@ Find out everything about the taxonomy: all Contexts and Events are open and doc
 [Check out the Reference](./reference/overview.md)
 
 :::info join the discussion
-Objectiv and the open taxonomy are open-source and we're building it in public. Have opinions on where we should take this next? [Join us on Slack](https://join.slack.com/t/objectiv-io/shared_invite/zt-u6xma89w-DLDvOB7pQer5QUs5B_~5pg)
+Objectiv and the open analytics taxonomy are open-source and we're building it in public. Have opinions on where we should take this next? [Join us on Slack](https://join.slack.com/t/objectiv-io/shared_invite/zt-u6xma89w-DLDvOB7pQer5QUs5B_~5pg)
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/introduction.md:36]: nice ones

File: docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md [hunk 12]
```diff
@@ -1,24 +1,35 @@
 # ApplicationLoadedEvent
 
-A [NonInteractiveEvent](/taxonomy/reference/events/NonInteractiveEvent.md) that is emitted after an application (eg. SPA) has finished loading. It contains a 
-[SectionContext](/taxonomy/reference/location-contexts/SectionContext.md).
+A [NonInteractiveEvent](/taxonomy/reference/events/NonInteractiveEvent.md) that is emitted after an application (eg. SPA) has finished loading.
```
[the remark this answers is not in the document]
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/events/ApplicationLoadedEvent.md:3]: agreed

File: docs/docs/taxonomy/reference/events/VisibleEvent.md (hunk 7 above)
Comment (vincenthoogsteder) [reply docs/docs/taxonomy/reference/events/VisibleEvent.md:3]: sharp

**Comment by sdirosa** [comment]
Did another round of manual review and found MANY small mistakes in taxonomy, core concepts, reference - especially in JSON and example snippets. 

I think all links and mentions of Taxonomy concepts are correct now.

**Pull request merged**